CONTRIBUTING.md

```diff
@@ -1,77 +1,304 @@
-# Contributing to convert2rhel
+# Contributing to Convert2RHEL
 
-## Coding guidelines
-1. All python code must be python 2.6/2.7/3.6 compatible
-1. The code should follow linting from pylint
+The following is a set of guidelines for contributing to Convert2RHEL codebase, which are hosted in
+the [OAMG Organization](https://github.com/oamg) on GitHub. These are mostly guidelines, not rules.
 
-## Developing locally
+<div align="center">
+
+### Table Of Contents
+
+</div>
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+
+* [Working with GitHub](#working-with-github)
+* [Setting up Git](#setting-up-git)
+* [Forking a repository](#forking-a-repository)
+* [Collaborating with Pull Requests](#collaborating-with-pull-requests)
+
+[Getting started with development](#getting-started-with-development)
+
+* [Dependencies for local development](#dependencies-for-local-development)
+* [Setting up the environment](#setting-up-the-environment)
+* [Writing tests](#writing-tests)
+
+[Styleguide](#styleguide)
+
+* [Git commit messages](#git-commit-message)
+* [Python code styleguide](#python-code-styleguide)
+* [Python documentation styleguide](#python-documentation-styleguide)
+
+[Additional Notes](#additional-information)
+
+* [Building rpms locally](#building-rpms-locally)
+* [Building copr builds locally](#building-copr-builds-locally)
```

Comment:
GitHub already has a TOC built-in in the markdown visualization, should we keep it in the document or remove it to use the own TOC provided by GitHub? 

![image](https://user-images.githubusercontent.com/2881357/143050075-41a9631e-b747-4743-9994-2b2eadd8df49.png)

Reply:
@SpyTec what do you think about that?

Reply:
Sounds good to me